Test: Primary Tag Widget Test 추가

WantOscar opened the pull request:
## 작업내용
- Tag Widget의 파일 이름을 Primary Tag Widget으로 변경.

- Primary Tag Widget에 대한 위젯 테스트를 추가함.
  -  property로 받아온 텍스트가 전달되어 렌더링 되는지 확인.
  -  텍스트의 style 검증.
  -  텍스트를 감싸는 컨테이너가 렌더링 되는지 확인.
  -  컨테이너의 boxDecoration, border 및 child의 패딩을 검증.
  -  property로 받아온 텍스트에 빈 문자열이 들어가도 렌더링 되는지 확인.

- Primary Tag Widget의 골든 테스트 추가함.

- 상욱님의 코드를 병합 후 충돌이 없음을 확인함.

++ 코멘트에서 수정이 필요하다 하여 수정 및 추가함.
- place card widget 내 하드코딩 => 외부로 받도록 property 추가.
  - 타이틀, 콘텐트, 뱃지 텍스트, 태그의 정상적인 렌더링 확인.
  - 태그가 null일 때 안전하게 아무 태그도 표시되지 않는지 확인.
  - 태그가 빈 리스트일 때 ui가 안전한지 확인.

- env 생성.

- merge with upstream develop.



## 테스트
- primary tag widget test 추가 후 test에 이상 없는지 확인.
<img width="1906" height="222" alt="image" src="https://github.com/user-attachments/assets/d6216952-9d26-434c-97e5-97527c9086c4" />

- golden test 추가 후 이미지 파일 확인.
<img width="800" height="600" alt="image" src="https://github.com/user-attachments/assets/8a47a249-072e-4cee-bb7c-8d7da66faef0" />

- place card widget의 widget test 추가.
<img width="1826" height="224" alt="image" src="https://github.com/user-attachments/assets/19100a99-a1d4-479d-9eaa-d85ce8a73f10" />

-golden test 추가 후 이미지 파일 확인.
<img width="800" height="600" alt="image" src="https://github.com/user-attachments/assets/54c32e33-4a1d-4ec5-b7a2-20ff0abbe094" />

WantOscar committed:
tag widget 파일 이름 primary tag widget으로 변경 및 test 완료

WantOscar committed:
Merge branch 'develop' into feature/main

feature/main merge with develop branch

WantOscar committed:
primary tag widget test 파일 golden test 완료

SangWook16074 commented on lib/src/features/home/presentation/widgets/place_card.dart:
```diff
@@ -58,9 +58,9 @@ class PlaceCard extends StatelessWidget {
                   PrimaryBadgeWidget(text: "용산구", horizontal: 8, vertical: 3),
                   SizedBox(width: 4),
-                  TagWidget(text: '#커피향 가득'),
+                  PrimaryTagWidget(text: '#커피향 가득'),
```
내부 하드코딩 말고 외부에서 받도록 바꿔야됩니다

SangWook16074 commented on lib/src/features/rank/presentation/widget/review_card_widget.dart:
```diff
@@ -83,7 +83,7 @@ class ReviewCardWidget extends StatelessWidget {
                         return const Padding(
                           padding: EdgeInsets.only(right: 4.0),
-                          child: TagWidget(text: '#커피향 가득'),
+                          child: PrimaryTagWidget(text: '#커피향 가득'),
```
마찬가지입니다

WantOscar replied:
수정해서 커밋 쳤습니다.

SangWook16074 commented on pubspec.yaml:
```diff
@@ -81,7 +81,6 @@ flutter:
 
   # To add assets to your application, add an assets section, like this:
   assets:
-    - .env
```
제대로 병합되지 않은거 같은데 확인이 필요합니다

WantOscar replied:
.env 파일이 필요한건가요? test에 오류 원인이 .env래서 지웠습니다.

SangWook16074 replied:
.env가 있어야 됩니다. .env가 오류원인이면 생성해서 테스트를 수행해야합니다.

WantOscar committed:
place card widget 내 하드코딩 => 외부로 받도록 property 추가.
place card widget test 추가.
place card widget golden test 추가.

SangWook16074 reviewed (changes requested):
.env를 제외하고 나머지 코드는 문제 없습니다. 골든 파일의 실패결과가 conflict를 이르키고 있는데, 해당 PR에서 골든파일을 직접 삭제할 수 가 없네요. 로컬환경에서 직접 찾아서 제거해야됩니다.

.env는 서버환경과 관련된 환경변수가 저장되어 있습니다. Slack 참조해보세요.

SangWook16074 commented:
병합 충돌은 해결했으나, .env는 해결해야합니다.

WantOscar committed:
env 생성

WantOscar committed:
merge with upstream develop

Merge branch 'develop' of https://github.com/WantOscar/solo_play_applicaiton into feature/main
